cmd/kubeadm/app/util/runtime/runtime.go

```diff
@@ -97,15 +97,23 @@ func (runtime *CRIRuntime) ListKubeContainers() ([]string, error) {
 func (runtime *CRIRuntime) RemoveContainers(containers []string) error {
 	errs := []error{}
 	for _, container := range containers {
-		out, err := runtime.crictl("stopp", container).CombinedOutput()
-		if err != nil {
-			// don't stop on errors, try to remove as many containers as possible
-			errs = append(errs, errors.Wrapf(err, "failed to stop running pod %s: output: %s, error", container, string(out)))
-		} else {
+		var lastErr error
+		for i := 0; i < constants.RemoveContainerRetry; i++ {
+			out, err := runtime.crictl("stopp", container).CombinedOutput()
+			if err != nil {
+				lastErr = errors.Wrapf(err, "failed to stop running pod %s: output: %s", container, string(out))
+				continue
+			}
 			out, err = runtime.crictl("rmp", container).CombinedOutput()
 			if err != nil {
-				errs = append(errs, errors.Wrapf(err, "failed to remove running container %s: output: %s, error", container, string(out)))
+				lastErr = errors.Wrapf(err, "failed to remove running container %s: output: %s", container, string(out))
+				continue
 			}
+			break
```

Comment:
```suggestion
			lastErr = nil
			break
```
if both stop & remove succeed, we may reset the lastErr.

Reply:
Good catch! Thanks.